### PR TITLE
CI: improvements & Performance fixes

### PR DESCRIPTION
**Pull request by Nasfame**
Adding proposed solution in https://github.com/bxcodec/dbresolver/issues/20 for global context to avoid unnecessary function calls.

Since context.Background is returning a global variable its fine to use backgroundCtx

### CODE REVIEW

**Comment by Nasfame** [comment]
@bxcodec opening a fresh PR!!

File: db.go [hunk 1]
```diff
@@ -9,6 +9,8 @@ import (
 	"go.uber.org/multierr"
 )
 
+var backgroundCtx = context.Background() //reduces unnecessary cross library calls
```
Comment (bxcodec) [inline db.go:12]: no, let's not use this. 
this makes backgroundCtx become global. As you know context contains states, and can be canceled. if one is cancelled, all context will be cancelled

Comment (Nasfame) [reply db.go:12]: Inside of context.Background()

image


Its just returning a global variable

Comment (Nasfame) [reply db.go:12]: This ctx cannot be cancelled or nor is it susceptible to cascading cancelations

**Review by Nasfame — COMMENTED** [review]
This pr is ready

**Comment by Nasfame** [comment]
Maintaining #33  and #32 has become cumbersome.

**Pull request closed without merging**